### PR TITLE
add BN

### PR DESCRIPTION
**Pull request by SiYue0211**
增加 batch normalization
删除点乘，用tf.multiply替代
base.py deconv_module deconv通道数改为1024

### CODE REVIEW

File: network/base.py [hunk 1]
```diff
@@ -197,57 +214,50 @@ def deconv_fc(self, d_i, d_o, input, name=None, trainable=True):
...
     # TODO batch norm
     @layer
-    def batch_normalize(self):
-        """
-            x_shape = x.get_shape()
-    params_shape = x_shape[-1:]
-
-    if c['use_bias']:
-        bias = _get_variable('bias', params_shape,
+    def batch_normalize(self, input):
```
Comment (cshjarry) [inline network/base.py]: 这个地方有现成api吧

Comment (cshjarry) [reply network/base.py]:
```python
    @layer
    def batch_normalization(self, input, name, relu=True, is_training=False):
        """contribution by miraclebiu"""
        if relu:
            temp_layer = tf.contrib.layers.batch_norm(input, scale=True, center=True, is_training=is_training,
                                                      scope=name)
            return tf.nn.relu(temp_layer)
        else:
            return tf.contrib.layers.batch_norm(input, scale=True, center=True, is_training=is_training, scope=name)

```

这个是现成的api你看下有没有什么不一样，或者是达不到我们的效果

File: network/base.py [hunk 2]
```diff
@@ -139,16 +162,13 @@ def deconv_module(self, name=None, module_num=None):
...
+         .conv(3, 3, 1024, 1, 1, name=name + '_conv2')
          .batch_normalize().relu()
-         .conv(3, 3, 512, 1, 1, name=name + 'conv3')
+         .conv(3, 3, 1024, 1, 1, name=name + 'conv3')
          .batch_normalize(name=name + '_feature'))
 
-        self.feed(name + '_deconv_output', name + '_feature') \
-            .eltw_prod().relu(name=name)
-        # the deconv module output will be put in the [name]
+        return tf.multiply(name + '_deconv_output', name + '_feature').relu(name=name)
```
Comment (cshjarry) [inline network/base.py]: 这个地方需要把两个层的数据取出，tensorflow函数只能接受tensor作为变量

**Comment by SiYue0211** [comment]
在训练时 ，参数COMMON TRAINING: True， 测试时改为Fasle
并且计算loss时要把updata_ops到train_ops中，这样才能让μ和σ的滑动平均
``
  update_ops = tf.get_collection(tf.GraphKeys.UPDATE_OPS)
  with tf.control_dependencies(update_ops):
    train_op = optimizer.minimize(loss)
``
